dbms/src/AggregateFunctions/AggregateFunctionAvgWeighted.cpp

```diff
@@ -0,0 +1,47 @@
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <AggregateFunctions/AggregateFunctionAvgWeighted.h>
+#include <AggregateFunctions/Helpers.h>
+#include <AggregateFunctions/FactoryHelpers.h>
+
+namespace DB
+{
+
+namespace
+{
+
+template <typename T>
+struct AvgWeighted
+{
+    using FieldType = std::conditional_t<IsDecimalNumber<T>, Decimal128, NearestFieldType<T>>;
+    using Function = AggregateFunctionAvgWeighted<T, AggregateFunctionAvgWeightedData<FieldType>>;
+};
+
+template <typename T>
```

Comment:
Now it is impossible to use different types for values and weights. It may be not very convenient.
it is possible to improve later.

Reply:
@KochetovNicolai ,
We are wanting to use the avgWeighted function but with custom weights for different fields. Would be great if that improvement can be provided.